### ISSUE TITLE
knife should use `ui.msg` instead of `puts`

### ISSUE THREAD

**Issue by zdenko-s** [chef/knife-ec2#415]
If you try to launch EC2 instance from Ruby code and redirect UI output to stream, EC2 commands are  printing output to screen directly. Very few outputs are redirected to stream.
How to reproduce:
`    Chef::Knife::Ec2ServerCreate.load_deps
    knife = Chef::Knife::Ec2ServerCreate.new()
    knife.configure_chef
   #set required params
# 

```
# setup output capture
stdout = StringIO.new
stderr = StringIO.new
knife.ui = Chef::Knife::UI.new(stdout, stderr, STDIN, {})
knife.run()
```

`
Output goes to screen.
I have checked source and problem is in **ec2_base**
```
    def msg_pair(label, value, color=:cyan)
        if value && !value.to_s.empty?
          puts "#{ui.color(label, color)}: #{value}"
        end
    end
```

Message is directly printed (puts). It should be printed using methods of ui. ( ui.info() )
It is done correctly later in code
```
if errors.each{|e| ui.error(e)}.any?
```

Errors are printed to STDERR stream. I tested it by invoking code with missing parameters.

**Comment by NimishaS** [comment]
@zdenko-s , how are you redirecting UI output to stream? Are you using `log_location` in `knife.rb` file?

Can you please share your usecase too?

**Comment by zdenko-s** [comment]
I am not modifying knife.rb. I have custom code where I instantiate knife plugin which is wrapped by command line "knife ec2 server create". Digging into source I realized that I have to use Ec2ServerCreate object and work with it.
Code I posted is from my Rails application. Code configures options and invokes method "run' (If I remember correctly). As I described in message, problem is ec2_base. Text is printed using method "puts" which prints to console. It should be printed using methods of ui object, which correctly redirects.
